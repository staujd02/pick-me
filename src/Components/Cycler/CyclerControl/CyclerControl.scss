@import '../../../Assets/styling/mixins.scss';
@import '../../../Assets/styling/palette.scss';

#cycler-controls {
    @include alpha-background-color(rgba($color1, $light), $color2);

    position: absolute;
    left: calc(50vw - 150px / 2);
    top: -25px;
    padding-top: 65px;
    height: 75px;
    width: 165px;
    border-radius: 25px;
    display: flex;
    flex-wrap: nowrap;
    flex-direction: row;
    justify-content: center;
    align-items: flex-end;
}

#cycler-controls button {
    @include alpha-background-color(rgba($color1, $heavy), $color2);
    @include button-shadow-effect;

    flex: 1 1;
    color: $color2;
    border: none;
    max-height: 50px;
    max-width: 50px;
    border-radius: 50px;
    cursor: pointer;
    margin: 10px;
}