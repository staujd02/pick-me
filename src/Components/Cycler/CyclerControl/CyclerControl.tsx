import React from 'react';
import './CyclerControl.scss';
import IconButton from '@material-ui/core/IconButton';
import SvgIcon from '@material-ui/core/SvgIcon';
import { PlayBackMode } from '../../../enumerations';

class CyclerControl extends React.Component<CyclerControlProps, CyclerControlState> {

    render(){
        const { onSkip, onPlayBack, playBackMode } = this.props;
        return (
            <div className='cycler-controls'>
                <IconButton id="skip-control" onClick={onSkip} aria-label="skip">
                    <SvgIcon xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
                        <path d="M12 36l17-12-17-12v24zm20-24v24h4V12h-4z" />
                    </SvgIcon>
                </IconButton>
                <IconButton id="playback-control" onClick={onPlayBack} aria-label={playBackMode}>
                    {playBackMode === PlayBackMode.Play && 
                        <SvgIcon xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
                            <path d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z" />
                        </SvgIcon>}
                    {playBackMode === PlayBackMode.Pause &&
                        <SvgIcon xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
                            <path d="M18 32h4V16h-4v16zm6-28C12.95 4 4 12.95 4 24s8.95 20 20 20 20-8.95 20-20S35.05 4 24 4zm0 36c-8.82 0-16-7.18-16-16S15.18 8 24 8s16 7.18 16 16-7.18 16-16 16zm2-8h4V16h-4v16z" />
                        </SvgIcon>}
                </IconButton>
            </div>);
    }

}

export default CyclerControl;