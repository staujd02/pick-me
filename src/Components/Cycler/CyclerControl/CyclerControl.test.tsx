import React from 'react';
import { ReactWrapper, mount } from "enzyme";
import { formatHTML } from '../../../TestUtilities/htmlFormatter';
import CyclerControl from './CyclerControl';

describe('The Cycler', () => {

    let wrapper: ReactWrapper<CyclerControlProps, CyclerControlState>;

    beforeEach(() => {
        wrapper = mount(<CyclerControl />)
    });

    it('renders correctly', () => expect(formatHTML(wrapper.html())).toMatchSnapshot());
});