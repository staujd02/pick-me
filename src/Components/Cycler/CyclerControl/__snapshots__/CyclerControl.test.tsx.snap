// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Cycler renders correctly 1`] = `
"
<div class=\\"cycler-controls\\">
  <button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" id=\\"skip-control\\" aria-label=\\"skip\\">
    <span class=\\"MuiIconButton-label\\">
      <svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 48 48\\" aria-hidden=\\"true\\" role=\\"presentation\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"48\\" height=\\"48\\">
        <path d=\\"M12 36l17-12-17-12v24zm20-24v24h4V12h-4z\\"></path>
      </svg>
    </span>
    <span class=\\"MuiTouchRipple-root\\"></span>
  </button>
  <button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" id=\\"playback-control\\">
    <span class=\\"MuiIconButton-label\\"></span>
    <span class=\\"MuiTouchRipple-root\\"></span>
  </button>
</div>
"
`;
